src/config.rs

```diff
@@ -45,68 +45,41 @@ pub(crate) struct NodeConfig {
-                );
-
-                if self.advanced.executor {
-                    node = node.system_domain(
-                        domains::ConfigBuilder::new().core(ConfigBuilder::new().build()),
-                    );
-                }
-                node = node.role(Role::Authority);
-                let chain_spec = chain_spec::gemini_3c()
-                    .expect("cannot extract the gemini3c chain spec from SDK");
-
-                node.build(self.directory, chain_spec).await.map_err(color_eyre::Report::msg)
-            }
-            ChainConfig::Dev => {
-                node = Node::dev();
-
-                if self.advanced.executor {
-                    node = node.system_domain(
-                        domains::ConfigBuilder::new().core(ConfigBuilder::new().build()),
-                    );
-                }
-                node = node.role(Role::Authority);
-                let chain_spec = chain_spec::gemini_3c()
-                    .expect("cannot extract the gemini3c chain spec from SDK");
-
-                node.build(self.directory, chain_spec).await.map_err(color_eyre::Report::msg)
-            }
-            ChainConfig::DevNet => {
-                let mut node = Node::devnet()
-                    .network(NetworkBuilder::devnet().name(self.name))
-                    .dsn(DsnBuilder::devnet().provider_storage_path(provider_storage_dir_getter()));
-
-                if self.advanced.executor {
-                    node = node.system_domain(
-                        domains::ConfigBuilder::new().core(ConfigBuilder::new().build()),
-                    );
-                }
-                node = node.role(Role::Authority);
-                let chain_spec = chain_spec::gemini_3c()
-                    .expect("cannot extract the gemini3c chain spec from SDK");
-
-                node.build(self.directory, chain_spec).await.map_err(color_eyre::Report::msg)
-            }
+        let mut node = match chain {
+            ChainConfig::Gemini3c => Node::gemini_3c()
+                .network(NetworkBuilder::gemini_3c().name(self.name))
+                .dsn(DsnBuilder::gemini_3c().provider_storage_path(provider_storage_dir_getter())),
+            ChainConfig::Dev => Node::dev(),
+            ChainConfig::DevNet => Node::devnet()
+                .network(NetworkBuilder::devnet().name(self.name))
+                .dsn(DsnBuilder::devnet().provider_storage_path(provider_storage_dir_getter())),
+        };
+
+        if self.advanced.executor {
+            node = node
+                .system_domain(domains::ConfigBuilder::new().core(ConfigBuilder::new().build()));
         }
+
+        node = node.role(Role::Authority);
+
+        let chain_spec =
+            chain_spec::gemini_3c().expect("cannot extract the gemini3c chain spec from SDK");
+
+        node.build(self.directory, chain_spec).await.map_err(color_eyre::Report::msg)
```

Comment:
It is the best to add context each time. Errors are like onions. Each layer should add more context to user. `I/O error: File doesn't exist` isn't that helpfull, while `Failed to build node`, `Failed to open provider storage`, `I/O error: File doesn't exist` is actually meaningfull.

![onion](https://i.ytimg.com/vi/zUQ9bjBfiRk/maxresdefault.jpg)
```suggestion
        node.build(self.directory, chain_spec).await.context("Failed to build node")
```